test/gtest/gather.hpp

```diff
@@ -0,0 +1,180 @@
+{
+    return {
+        GatherTestCase({5, 6, 7}, {5, 6, 7}, 0),         // same shape
+        GatherTestCase({15, 30, 45}, {15, 30, 45}, 2),   // same shape
+        GatherTestCase({2, 2, 3}, {2, 2, 2}, 1),         // different shape
+        GatherTestCase({16, 16, 3}, {24, 2, 2}, 0),      // different shape
+        GatherTestCase({16, 256, 256}, {16, 256, 16}, 1) // large size
+    };
+}
+
+template <typename T>
+struct GatherFwdTest : public ::testing::TestWithParam<GatherTestCase>
+{
+protected:
+    void SetUp() override
+    {
+        auto&& handle = get_handle();
+        config        = GetParam();
+
+        auto input_dims   = config.GetInputDim();
+        auto indices_dims = config.GetIndicesDim();
+        dim               = config.GetDim();
+
+        std::vector<size_t> output_dims = indices_dims;
+
+        gatherDesc.setMode(MIOPEN_GATHER);
+        gatherDesc.setDim(dim);
+
+        auto gen_value = [](auto...) { return prng::gen_descreet_uniform_sign<T>(1e-2, 100); };
+        auto dim_size  = input_dims[dim];
+        auto gen_index = [dim_size](auto...) {
+            return prng::gen_0_to_B(static_cast<int64_t>(dim_size));
+        };
+
+        indices = tensor<int64_t>{indices_dims}.generate(gen_index);
+        input   = tensor<T>{input_dims}.generate(gen_value);
+
+        output = tensor<T>{output_dims};
+        std::fill(output.begin(), output.end(), static_cast<T>(0));
+
+        ref_output = tensor<T>{output_dims};
+        std::fill(ref_output.begin(), ref_output.end(), static_cast<T>(0));
+
+        input_dev   = handle.Write(input.data);
+        indices_dev = handle.Write(indices.data);
+        output_dev  = handle.Write(output.data);
+    }
+
+    void RunTest()
+    {
+        auto&& handle = get_handle();
+
+        cpu_gather_forward<T>(input, indices, ref_output, dim);
+
+        miopenStatus_t status;
+        status = gatherDesc.Forward(handle,
+                                    input.desc,
+                                    input_dev.get(),
+                                    indices.desc,
+                                    indices_dev.get(),
+                                    output.desc,
+                                    output_dev.get());
+
+        ASSERT_EQ(status, miopenStatusSuccess);
+
+        output.data = handle.Read<T>(output_dev, output.data.size());
+    }
+
+    double GetTolerance()
+    {
+        double tolerance = std::numeric_limits<T>::epsilon() * 10;
+        return tolerance;
+    }
+
+    void Verify()
+    {
+        double threshold = GetTolerance();
+        auto error       = miopen::rms_range(ref_output, output);
+        EXPECT_LT(error, threshold) << "Error output (param grad) beyond tolerance Error:" << error
+                                    << ",  Threshold: " << threshold << std::endl;
+    }
+
+    GatherTestCase config;
+
+    miopen::GatherDescriptor gatherDesc;
+    tensor<T> input;
+    tensor<T> output;
+    tensor<int64_t> indices;
```

Comment:
same as above: consider to use `uint64_t` instead of `int64_t`

Reply:
updated.